### ISSUE TITLE
Package search location problem (System.InvalidOperationException: Failed to resolve the following dependencies for target framework 'DNX,Version=v4.5.1')

### ISSUE THREAD

**Issue by buraktamturk** [aspnet/dnx#2454]
Hello,

This could be duplicate of #1750 or other issues. However they do not mention the problem clearly (and sadly they are closed), I had to open a new one. I wonder wherever this belongs dnx, dnvm or home.

I use VS 2015, beta6 vnext libraries alongside beta6 dnx and I deploy to IIS using VS.

I get the exception

```
System.InvalidOperationException: Failed to resolve the following dependencies for target framework 'DNX,Version=v4.5.1':
```

Mentions many packages that could not be found at these locations.

```
Searched Locations:
 C:\inetpub\my-project-name\approot\src\{name}\project.json
 C:\Windows\system32\config\systemprofile\.dnx\packages\{name}\{version}\{name}.nuspec
 C:\Windows\Microsoft.NET\assembly\GAC_32\{name}\{version}\{name}.dll
 C:\Windows\Microsoft.NET\assembly\GAC_64\{name}\{version}\{name}.dll
 C:\Windows\Microsoft.NET\assembly\GAC_MSIL\{name}\{version}\{name}.dll
```

My project is located at C:\inetpub\my-project-name, and these packages are located at C:\inetpub\my-project-name\approot\packages, which is not in the Searched locations.

If I copy whole folder contents C:\inetpub\my-project-name\approot\packages to C:\Windows\system32\config\systemprofile\.dnx\packages, the project works. But I had to to this everytime I add/update the libraries.

Are there any particular reason the folder ("C:\inetpub\my-project-name\approot\packages") is not included in the Searched Locations? Are there any workarounds for it?

**It started to happen in beta6 version**

Full exception:

```
Exception Details: System.InvalidOperationException: Failed to resolve the following dependencies for target framework 'DNX,Version=v4.5.1':
  EntityFramework 6.1.3
  Kestrel 1.0.0-beta6
  Microsoft.AspNet.Antiforgery 1.0.0-beta6
  Microsoft.AspNet.Authorization 1.0.0-beta6
  Microsoft.AspNet.Cors 1.0.0-beta6
  Microsoft.AspNet.Cors.Core 1.0.0-beta6
  Microsoft.AspNet.Cryptography.Internal 1.0.0-beta6
  Microsoft.AspNet.DataProtection 1.0.0-beta6
  Microsoft.AspNet.DataProtection.Abstractions 1.0.0-beta6
  Microsoft.AspNet.Diagnostics 1.0.0-beta6
  Microsoft.AspNet.Diagnostics.Abstractions 1.0.0-beta6
  Microsoft.AspNet.FeatureModel 1.0.0-beta6
  Microsoft.AspNet.FileProviders.Abstractions 1.0.0-beta6
  Microsoft.AspNet.FileProviders.Physical 1.0.0-beta6
  Microsoft.AspNet.Hosting 1.0.0-beta6
  Microsoft.AspNet.Hosting.Abstractions 1.0.0-beta6
  Microsoft.AspNet.Hosting.Server.Abstractions 1.0.0-beta6
  Microsoft.AspNet.Html.Abstractions 1.0.0-beta6
  Microsoft.AspNet.Http 1.0.0-beta6
  Microsoft.AspNet.Http.Abstractions 1.0.0-beta6
  Microsoft.AspNet.Http.Extensions 1.0.0-beta6
  Microsoft.AspNet.Http.Features 1.0.0-beta6
  Microsoft.AspNet.JsonPatch 1.0.0-beta6
  Microsoft.AspNet.Loader.IIS 1.0.0-beta6
  Microsoft.AspNet.Loader.IIS.Interop 1.0.0-beta6
  Microsoft.AspNet.Localization 1.0.0-beta6
  Microsoft.AspNet.Mvc 6.0.0-beta6
  Microsoft.AspNet.Mvc.Abstractions 6.0.0-beta6
  Microsoft.AspNet.Mvc.ApiExplorer 6.0.0-beta6
  Microsoft.AspNet.Mvc.Core 6.0.0-beta6
  Microsoft.AspNet.Mvc.Cors 6.0.0-beta6
  Microsoft.AspNet.Mvc.DataAnnotations 6.0.0-beta6
  Microsoft.AspNet.Mvc.Formatters.Json 6.0.0-beta6
  Microsoft.AspNet.Mvc.Localization 6.0.0-beta6
  Microsoft.AspNet.Mvc.Razor 6.0.0-beta6
  Microsoft.AspNet.Mvc.Razor.Host 6.0.0-beta6
  Microsoft.AspNet.Mvc.ViewFeatures 6.0.0-beta6
  Microsoft.AspNet.PageExecutionInstrumentation.Interfaces 1.0.0-beta6
  Microsoft.AspNet.Razor 4.0.0-beta6
  Microsoft.AspNet.Razor.Runtime 4.0.0-beta6
  Microsoft.AspNet.Routing 1.0.0-beta6
  Microsoft.AspNet.Server.IIS 1.0.0-beta6
  Microsoft.AspNet.Server.Kestrel 1.0.0-beta6
  Microsoft.AspNet.Server.WebListener 1.0.0-beta6
  Microsoft.AspNet.StaticFiles 1.0.0-beta6
  Microsoft.AspNet.WebApi.Client 5.2.3
  Microsoft.AspNet.WebUtilities 1.0.0-beta6
  Microsoft.CodeAnalysis.Analyzers 1.0.0
  Microsoft.CodeAnalysis.Common 1.0.0
  Microsoft.CodeAnalysis.CSharp 1.0.0
  Microsoft.Framework.Caching.Abstractions 1.0.0-beta6
  Microsoft.Framework.Caching.Memory 1.0.0-beta6
  Microsoft.Framework.Configuration 1.0.0-beta6
  Microsoft.Framework.Configuration.Abstractions 1.0.0-beta6
  Microsoft.Framework.Configuration.Binder 1.0.0-beta6
  Microsoft.Framework.Configuration.CommandLine 1.0.0-beta6
  Microsoft.Framework.Configuration.EnvironmentVariables 1.0.0-beta6
  Microsoft.Framework.Configuration.Ini 1.0.0-beta6
  Microsoft.Framework.Configuration.Json 1.0.0-beta6
  Microsoft.Framework.DependencyInjection 1.0.0-beta6
  Microsoft.Framework.DependencyInjection.Abstractions 1.0.0-beta6
  Microsoft.Framework.Globalization.CultureInfoCache 1.0.0-beta6
  Microsoft.Framework.Localization 1.0.0-beta6
  Microsoft.Framework.Localization.Abstractions 1.0.0-beta6
  Microsoft.Framework.Logging 1.0.0-beta6
  Microsoft.Framework.Logging.Abstractions 1.0.0-beta6
  Microsoft.Framework.Logging.Console 1.0.0-beta6
  Microsoft.Framework.Notification 1.0.0-beta6
  Microsoft.Framework.OptionsModel 1.0.0-beta6
  Microsoft.Framework.Runtime.Abstractions 1.0.0-beta6
  Microsoft.Framework.Runtime.Roslyn.Abstractions 1.0.0-beta6
  Microsoft.Framework.Runtime.Roslyn.Common 1.0.0-beta6
  Microsoft.Framework.WebEncoders 1.0.0-beta6
  Microsoft.Framework.WebEncoders.Core 1.0.0-beta6
  Microsoft.Net.Http.Headers 1.0.0-beta6
  Microsoft.Net.Http.Server 1.0.0-beta6
  Microsoft.Net.WebSockets 1.0.0-beta6
  Microsoft.VisualStudio.Web.BrowserLink.Loader 14.0.0-beta6
  Newtonsoft.Json 6.0.6
  Refit 2.2.1
  System.Collections.Immutable 1.1.36
  System.Reflection.Metadata 1.0.21

Searched Locations:
 C:\inetpub\my-project-name\approot\src\{name}\project.json
 C:\Windows\system32\config\systemprofile\.dnx\packages\{name}\{version}\{name}.nuspec
 C:\Windows\Microsoft.NET\assembly\GAC_32\{name}\{version}\{name}.dll
 C:\Windows\Microsoft.NET\assembly\GAC_64\{name}\{version}\{name}.dll
 C:\Windows\Microsoft.NET\assembly\GAC_MSIL\{name}\{version}\{name}.dll

Try running 'dnu restore'.
```

**Comment by moozzyk** [comment]
You can't use EF6 with Asp.Net 5

**Comment by Eilon** [comment]
@moozzyk why not? I think many people are using it...

**Comment by moozzyk** [comment]
For some reason I thought it was on CoreClr. Will remove my comment.

**Comment by buraktamturk** [comment]
@moozzyk There are downsides but I am currently using it as it fits perfectly for my needs. I don't use CoreCLR, and I wrote the models manually instead of using edmx

**Comment by davidfowl** [comment]
@buraktamturk How are you running on IIS? Did you do a `dnu publish`?

**Comment by buraktamturk** [comment]
I use "Publish" button on VS 2015, and deploy to server using Web Deploy 3.6. I also don't tick "deploy using powershell script" option, if I do I get SSL errors and I can't make it to accept it.

**Comment by buraktamturk** [comment]
I was using same technique when I was targeting beta-5 or beta-4, everything was great.

**Comment by buraktamturk** [comment]
I decided to go with beta-5 until this issue gets fixed, however I get same problem when I back to beta5 and I didn't have any problems before I upgrade the project to beta6.

When I switch the project beta-5 to beta-6 using Properties window in Visual Studio, VS has generated global.json using this content:

```javascript
{
  "sdk": {
    "version": "1.0.0-beta6",
    "runtime": "clr",
    "architecture": "x64"
  }
}
```

When I switch to beta5, the "version" in this json indeed changed (previously I didn't had global.json while using beta5, it is auto generated when I switch beta6). But there was same problem again.

So looked an example of global.json (from the internet) and modified the contents accordingly.

```javascript
{
  "projects": ["."],
  "packages": "packages",
  "sdk": {
    "version": "1.0.0-beta5",
    "runtime": "clr",
    "architecture": "x64"
  }
}
```

And boom! It works beta5, so I had to test with beta6:

```javascript
{
  "projects": ["."],
  "packages": "packages",
  "sdk": {
    "version": "1.0.0-beta6",
    "runtime": "clr",
    "architecture": "x64"
  }
}
```

Everything works great again. I am so happy!

**the problem was lack of "packages": "packages", in global.json**

I am sorry that I kept nice programmers busy with this issue, I think it's still an issue (at least we know the root cause) from my point of view but clearly not on the DNX part. (maybe at VS part?)
